common/components/register_model/spec.yaml

```diff
@@ -0,0 +1,63 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: register_model
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: Register model
+description:
+  Register a model to a workspace or a registry. The component works on compute with [MSI](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-create-manage-compute-instance?tabs=python) attached.
+
+environment: azureml:python-sdk-v2:1
+
+code: ../../src/model_registration
+command: >-
+  python register.py
+  --model_path ${{inputs.model_path}}
+  $[[--model_type ${{inputs.model_type}}]]
+  $[[--model_name ${{inputs.model_name}}]]
+  $[[--registry_name ${{inputs.registry_name}}]]
+  $[[--model_info_path ${{inputs.model_info_path}}]]
+  $[[--model_description "${{inputs.model_description}}"]]
+  --registration_details ${{outputs.registration_details}}
+
+inputs:
+  model_name:
+    type: string
+    optional: true
+    description: Model name to use in the registration. If name already exists, the version will be auto incremented
+
+  model_type:
+    type: string
+    optional: true
+    enum:
+      - custom_model
+      - mlflow_model
```

Comment:
How do we align and keep this up-to-date with the v2 sdk?

Reply:
Right now, have to update this manually if anything changes on v2 sdk front.